src/xcode/ENA/ENA/Source/Services/Exposure Submission/ExposureSubmissionServiceFactory.swift

```diff
@@ -38,14 +39,23 @@ extension ExposureSubmissionServiceFactory {
 #endif
 
 // MARK: UI Testing implementation.
-#if UITESTING
+#if DEBUG
 
 /// This extension will return a mock service if and only if the .useMock parameter is passed to the application.
 /// If the parameter is _not_ provided, the factory will instantiate a regular ENAExposureSubmissionService.
 /// - NOTE: This is condtionally compiled so no test code spills into the release build.
 extension ExposureSubmissionServiceFactory {
 	static func create(diagnosiskeyRetrieval: DiagnosisKeysRetrieval, client: Client, store: Store) -> ExposureSubmissionService {
 
+		// default implementation if in non UI testing mode
+		guard isUITesting else {
+			return ENAExposureSubmissionService(
+				diagnosiskeyRetrieval: diagnosiskeyRetrieval,
+				client: client,
+				store: store
+			)
```

Comment:
I see a problem with the `isUITesting` approach here (and in other places). You copied this code form above, and now in unit testing this return is tested, while in production, the return in line 31 is executed. That means we can't test the actual implementation in our unit test. It seems that in general, you had to copy code in multiple instances and the production code is not testable anymore in these places.

Reply:
I see the point and this problem (and maybe others) are a result of the current interpretation of the previous implementation. 

Previously there were `UITESTING` blocks for… well… ui testing only and the 'normal' code for prod, dev and whatever use cases. This results in such glorious chunks of code such as this ServiceFactory. This piece of code (and maybe others) is a 1:1 refactoring with the new `isUITesting` flag. Normally I would expect something like this:

```swift
// ignore the indentation! 
static func create(diagnosiskeyRetrieval: DiagnosisKeysRetrieval, client: Client, store: Store) -> ExposureSubmissionService {
  #if DEBUG
    if isUITesting {
      // UI test code
      return // skip rest of the code - if applicable
    } // `else` would lead to custom code for unit tests - if needed
  #endif
  […] // 'normal' code
}
```

I chose the default `DEBUG` flag to indicate 'test' or 'non-production' code. No need for `UITESTING`, `TESTING` or whatever. My biggest pain point with the extra `UITESTING` flag was, that we needed two targets/schemes for unit and ui testing. Both could be run sequentially in fastlane's `run_tests` (we just did the unit testing) but when it came to the code coverage [`xcov`](https://docs.fastlane.tools/actions/xcov/) only accepted one scheme to use the coverage from - which one to choose? This PR combines these two schemes and replaces an extra precompiler flag with a well known (DEBUG) one.

One final note. There are tons of alternatives how to control the UI in ui testing and how to inject mock data. Using [`Current`](https://www.pointfree.co/blog/posts/21-how-to-control-the-world) is a very nice one but requires a way bigger refactoring than this.

Reply:
I might refactor this whole file to this proposed structure. I skipped this during development to keep the amount of 'noise' from this PR as low as possible.